pyTigerGraph/pyTigerGraph.py

```diff
@@ -1339,96 +1435,94 @@ def getLicenseInfo(self):
             raise TigerGraphException(res["message"], res["code"])
         return ret
 
-
     # GSQL support =================================================
 
     def initGsql(self, jarLocation="~/.gsql", certLocation="~/.gsql/my-cert.txt"):
-        
+
         self.jarLocation = os.path.expanduser(jarLocation)
         self.certLocation = os.path.expanduser(certLocation)
-        self.url = self.gsUrl.replace("https://", "").replace("http://", "") # Getting URL with gsql port w/o https://
-
+        self.url = self.gsUrl.replace("https://", "").replace("http://", "")  # Getting URL with gsql port w/o https://
 
         # Check if java runtime is installed.
         if subprocess.run(['which', 'java']).returncode != 0:
             raise TigerGraphException("Could not find java runtime. Please download and install from https://www.oracle.com/java/technologies/javase-downloads.html", None)
 
         # Create a directory for the jar file if it does not exist.
-        if(not os.path.exists(self.jarLocation)):
+        if not os.path.exists(self.jarLocation):
             os.mkdir(self.jarLocation)
 
-        # Downlad the gsql_client.jar file
+        # Download the gsql_client.jar file
         if self.downloadJar:
             print("Downloading gsql client Jar")
 
-            jar_url = ('https://bintray.com/api/ui/download/tigergraphecosys/tgjars/' 
-                    + 'com/tigergraph/client/gsql_client/' + self.clientVersion 
-                    + '/gsql_client-' + self.clientVersion + '.jar')        
-            r = requests.get(jar_url) 
-            open(self.jarLocation + '/gsql_client.jar', 'wb').write(r.content)
-        
-        if self.downloadCert: #HTTP/HTTPS
+            ver = self.getVer()
```

Comment:
Seems like getVer() requires a token, which may not be present.

Reply:
This requirement should be checked (amongst other things) in the `__init__`, I believe, and fail at the very beginning instead of sometime later (e.g. when the gsql part would be first used).

Reply:
It doesn't have to fail at all. We can use getVer() only when we have a token and have a default otherwise.